packages/storage/src/postgres.rs

```diff
@@ -678,6 +685,136 @@ impl Postgres {
         Ok(())
     }
 
+    pub(crate) async fn _update_costs(
+        &self,
+        cost_per_tx: Vec<TransactionCostUpdate>,
```

Comment:
I'd change this fn so it is called for each `TransactionCostUpdate` separately. Reason being that now that you're no longer doing db transactions, one bundle could pass, another could revert and you'd be left in a partially updated state.

If you do a tx by tx then if it fails it won't update anything on the db side.

Reply:
Another suggestion to make it even more robust is to merge together the tx state updating and the cost updating into one method seeing as they're always called together in the state listener. 

That way you can wrap everything in one transaction and you cannot have a situation where you've updated the state but failed to update the costs. If anything fails you'll get another chance the next time the state listener runs.

The longer db tx locking time might be worth the retry-ability we'd get.